Drop-In/src/main/java/com/braintreepayments/api/SupportedPaymentMethodsFragment.java

```diff
@@ -22,6 +23,12 @@
 
 public class SupportedPaymentMethodsFragment extends Fragment implements SupportedPaymentMethodSelectedListener, VaultedPaymentMethodSelectedListener {
 
+    private enum ViewState {
```

Comment:
Can I get a little more info on what this ViewState is needed for? I feel like throughout the SDK we have avoided maintaining a state like this

Reply:
The main reason is to keep the view presentation logic in one place. Calling `setViewState` triggers a re-layout.

Reply:
Honestly thinking about leveraging the presenter pattern here to separate concerns. This view is complex because the loader being `visible` is driven by a lot of factors external to this view.

Reply:
On second thought, [this](https://github.com/braintree/braintree-android-drop-in/blob/4be12a3b3ad1bce680c52b8245bf5d811d4e6b5a/Drop-In/src/main/java/com/braintreepayments/api/SupportedPaymentMethodsFragment.java#L138) was the primary reason. When this view comes to the foreground we have to decide whether or not to show the loader. When DropIn is `FINISHING` we allow the loader to remain on the screen.